image-loader/app/controllers/ImageLoaderController.scala

```diff
@@ -244,6 +252,73 @@ class ImageLoaderController(auth: Authentication,
     }
   }
 
+  lazy val replicaS3: AmazonS3 = S3Ops.buildS3Client(config, maybeRegionOverride = Some("us-west-1"))
+
+  private case class RestoreFromReplicaForm(imageId: String)
+  def restoreFromReplica: Action[AnyContent] = AuthenticatedAndAuthorised.async { implicit request =>
+
+    val imageId = Form(
+      mapping(
+        "imageId" -> text
+      )(RestoreFromReplicaForm.apply)(RestoreFromReplicaForm.unapply)
+    ).bindFromRequest.get.imageId
+
+    implicit val logMarker: LogMarker = MarkerMap(
+      "imageId" -> imageId,
+      "requestType" -> "image-projection",
+      "requestId" -> RequestLoggingFilter.getRequestId(request)
+    )
+
+    config.maybeImageReplicaBucket match {
+      case _ if store.doesOriginalExist(imageId) =>
```

Comment:
maybe these need to be inside a future too? or maybe you can wrap the whole thing in one big future? you might end up with a Future[Future[_]] but if so you should be able to .flatten that out